From: Andrew Johnson <anj@aps.anl.gov>
Date: Thu, 19 Feb 2015 15:55:48 -0600
Subject: db: Allow link fields to return a DOUBLE value

A dbCa link does a ca_get with type DBR_CTRL_DOUBLE
to populate its list of attribute values immediately
after connecting. If the target is a DBF_*LINK field
it used to return an error, preventing the link from
properly connecting. This change makes dbGetField()
return a single NAN value instead of rejecting the
request.

Fixes: lp:545358
---
 src/db/dbAccess.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/db/dbAccess.c b/src/db/dbAccess.c
index 7b5415f..8fd5f73 100644
--- a/src/db/dbAccess.c
+++ b/src/db/dbAccess.c
@@ -962,6 +962,11 @@ long epicsShareAPI dbGetField(DBADDR *paddr,short dbrType,
             if (nRequest && *nRequest > 1) *nRequest = 1;
             break;
 
+        case DBR_DOUBLE:    /* Needed for dbCa links */
+            if (nRequest && *nRequest) *nRequest = 1;
+            *(double *)pbuffer = epicsNAN;
+            goto done;
+
         case DBR_CHAR:
         case DBR_UCHAR:
             if (nRequest && *nRequest > 0) {
